common/app/templates/inlineJS/nonBlocking/detectAdblock.scala.js

```diff
@@ -0,0 +1,28 @@
+try {
+    ((document, window) => {
+        window.guardian.adBlockers = {};
+        var ad = document.createElement('div');
+        ad.style.position = 'absolute';
+        ad.style.left = '-9999px';
+        ad.style.height = '10px';
+        ad.innerHTML = '&nbsp;';
+        ad.setAttribute('class', 'ad_unit');
+
+        // avoid a forced sync layout
```

Comment:
Long term, making adblock detection async is very helpful, because it opens up the way to more accurate, non-synchronous heuristics - like seeing if an ad request gets blocked at the network level.